sdk/storage/Azure.Storage.Blobs/tests/BlobTestBase.cs

```diff
@@ -523,5 +524,55 @@ public async ValueTask DisposeAsync()
                 }
             }
         }
+
+        public class DisposingFileStream : Stream
```

Comment:
Why not derive from `FileStream` and then you only need override `Dispose`?  I think you can use the [`FileStream.Name` property](https://docs.microsoft.com/en-us/dotnet/api/system.io.filestream.name?view=netframework-4.8) so you don't have to worry about tracking the path.

We should move this out into its own file and maybe call it something like `TemporaryFileStream` since all streams already dispose.